Automattic/jetpack
# Add: A few Jetpack notices and a framework for building more.

![screen shot 2016-05-22 at 12 16 50 pm](https://cloud.githubusercontent.com/assets/7129409/15455096/f448990a-2017-11e6-896a-d93e73815c4c.png)

The goal with `jetpack_notices` is to re-create all of the app-specific ones in React.  This is the first round of that. 

Lay out the framework for displaying jetpack-specific notices within the app. These changes include: 
- Changes the structure of the window.Initial_State.connectionStatus to include much more complete data about the connetion status
- Adds a JetpackNotices component that will conditionally show:
-- Dev Version notice
-- Dev Mode notice with specific text based on implmentation
-- Disconnect Notice, as an example to how we can trigger notices within the app.
- Removes some unnecessary Query Components.
- Removes some unnecessary PHP notices. 

All data for Jetpack Notices is initially retrieved through window.Initial_State

**To Test:** 
- You should see two notices after checking out.  One for dev version, which is not dismissible.  One that is a welcome message, that is dismissible.  
- The dismissible one should be dismissible. 
- The "What do you want to see on this page" notice at the bottom should be dismissible (but not styled)
- Enable dev mode in the following ways.  You should see correct messaging for each. 

`add_filter( 'jetpack_development_mode', '__return_true' );` (in a functionality plugin)

`define( 'JETPACK_DEV_DEBUG', true );` in wp-config.php

and also hack [this](https://github.com/Automattic/jetpack/compare/add/jetpack-notices?expand=1#diff-d8d5ed819214c3890a8dbe9ddceef2c7R135) to be true -- (url with no dot)

- Disconnect Jetpack.  You should see a notice with a link to the disconnect survey.  

This will need a little design love, but I'd like to get this in, as a base.
